utils/src/lib.rs

```diff
@@ -89,6 +94,35 @@ impl AllArtifactsBySlug {
     }
 }
 
+/// Links the given artifacts with a link key constructor function, passing the result of each
+/// linkage to the given callback.
+///
+/// This function will recursively link all artifacts until none are unlinked. It does this by:
+///
+/// 1. Using the specified predeployed library addresses (`deployed_library_addresses`) for known
+/// libraries (specified by the user) 2. Otherwise, computing the address the library would live at
+/// if deployed by `sender`, given a starting nonce of `nonce`.
+///
+/// If the library was already deployed previously in step 2., the linker will re-use the previously
```

Comment:
I'd remove the `.` here:
```suggestion
/// If the library was already deployed previously in step 2, the linker will re-use the previously
```